### ISSUE TITLE
aws_events_targets: Support AppSync GraphQL targets

### ISSUE THREAD

**Issue by thelegendtubaguy** [aws/aws-cdk#29884]
### Describe the feature

EventBridge now supports setting rule targets to an AppSync API:
https://aws.amazon.com/blogs/compute/build-real-time-applications-with-amazon-eventbridge-and-aws-appsync/

I would like to use CDK to configure my event bus rule to target an appsync api mutation.

### Use Case

When using an event based architecture, it would be extremely beneficial to route events matching a pattern to any number of subscribed clients to a graphql api.

### Proposed Solution

CloudFormation already has this capability, so potentially follow what you can do in CF.
https://serverlessland.com/patterns/eventbridge-to-appsync-sam

### Other Information

Will also need the ability to set the input transformer and pass appsync parameters.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.137.0

### Environment details (OS name and version, etc.)

Typescript 5.4.5

**Comment by pahud** [comment]
Yeah, looks like we need to implement AppSync GraphQL targets for the [aws-events-targets](https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk-lib/aws-events-targets/lib) library. It should be very straightforward, we just need to create a new class which implements `events.IRuleTarget` and return relevant config in the bind() method. Check the [EventBus](https://github.com/aws/aws-cdk/blob/91246acde1ab0512ea6b375f66c283516cb6f2b0/packages/aws-cdk-lib/aws-events-targets/lib/event-bus.ts#L35C34-L51) target for your reference.

Are you interested to submit a PR to help us move this forward?
